docs/spec/spec.md

```diff
@@ -0,0 +1,234 @@
+    * 3.4 [Compare operation](#34-compare-operation)
+    * 3.5 [Search operation](#35-search-operation)
+    * 3.6 [Search with type operation](#36-search-with-type-operation)
+    * 3.7 [Delete operation](#37-delete-operation)
+    * 3.8 [Close operation](#38-close-operation)
+4. [The `ldap:LdapResponse` type](#4-the-ldapldapresponse-type)
+5. [The `ldap:Error` type](#5-the-ldaperror-type)
+
+## 1. Overview
+
+The Ballerina LDAP module provides support for interacting with directory servers which support LDAP protocol. This module provided various directory operations by establishing a connection to an LDAP server, allowing for tasks such as adding, modifying, deleting, and searching directory entries. It focus on ease of use and integration, making it straightforward for developers to perform directory-based operations within their Ballerina applications.
+
+## 2. LDAP client
+
+The `ldap:Client` instance needs to be initialized before performing the functionalities. Once initialized, it can perform various operations on directories. Currently, it supports the generic LDAP operations; `add`, `modify`, `modifyDN`, `compare`, `search`, `searchWithType`, `delete`, and `close`.
+
+### 2.1 Configurations
+
+When initializing the `ldap:Client`, `ldap:ConnectionConfig` configuration can be provided.
+
+```ballerina
+# Provides a set of configurations to connect with a directory server.
+#
+# + hostName - The host name of the active directory server
+# + port -  The port of the active directory server
+# + domainName -  The domain name of the active directory
+# + password - The password of the active directory
+public type ConnectionConfig record {|
+    string hostName;
+    int port;
+    string domainName;
+    string password;
+|};
+```
+
+### 2.2 Initialization
+
+The `init` method initializes the `ldap:Client` instance using the parameters `hostName`, `port`, `domainName`, and `password`. The `hostName` and `port` parameters are used to bind the request and authenticate clients with the directory server, while the `domainName` and `password` parameters establish the connection to the server for performing LDAP operations. In case of failure, the method returns an `ldap:Error`."
+
+```ballerina
+# Gets invoked to initialize the LDAP client.
+#
+# + config - The configurations to be used when initializing the client
+# + return - A `ldap:Error` if client initialization failed
+public isolated function init(*ConnectionConfig config) returns Error?;
+```
+
+## 3. Operation types
+
+The currently supported operation types in LDAP are listed here.
+
+### 3.1 Add operation
+
+Creates an entry in a directory server.
+
+```ballerina
+# Creates an entry in a directory server.
+# 
+# + dN - The distinguished name of the entry
+# + entry - The information to add
+# + return - A `ldap:Error` if the operation fails or `ldap:LdapResponse` if successfully created
+remote isolated function add(string dN, Entry entry) returns LdapResponse|Error;
```

Comment:
Shall we add a section to describe what is `ldap:Entry` ?

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerina-ldap/pull/58/commits/d393c7dc09629ff0a3b5de1ba92727c9c9538dc8